extensions/kubernetes-client/runtime-internal/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesClientUtils.java

```diff
@@ -1,18 +1,48 @@
 package io.quarkus.kubernetes.client.runtime;
 
 import java.time.Duration;
+import java.util.Objects;
+import java.util.ServiceLoader;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
 
 import org.eclipse.microprofile.config.ConfigProvider;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.fabric8.kubernetes.api.model.KubernetesResource;
 import io.fabric8.kubernetes.client.Config;
 import io.fabric8.kubernetes.client.ConfigBuilder;
 import io.fabric8.kubernetes.client.KubernetesClient;
 import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+import io.fabric8.kubernetes.client.utils.KubernetesSerialization;
+import io.fabric8.kubernetes.internal.KubernetesDeserializer;
 import io.quarkus.runtime.TlsConfig;
 
 public class KubernetesClientUtils {
 
     private static final String PREFIX = "quarkus.kubernetes-client.";
+    private static final ObjectMapper KUBERNETES_CLIENT_MAPPER = new ObjectMapper();
```

Comment:
I very much want to avoid this, but I would need to time to look into alternatives

Reply:
So in order for me to understand: before this change, was the k8s client in Quarkus using it's own mapper or not?

Reply:
Looking at this more, we probably can't avoid is done here. What we do want to avoid however is using lambdas

Reply:
> So in order for me to understand: before this change, was the k8s client in Quarkus using it's own mapper or not?

The Kubernetes Client was using a **static** serialization utils class. Everything was static.

We're doing some changes to make the serialization part instantiatable, so you can have multiple clients with multiple serialization configurations.

Seems like the move to instances is not playing nice with SPI, native, and these processors.

Reply:
> The Kubernetes Client was using a static serialization utils class. Everything was static.

Good to know. Just to make sure there is no misunderstanding: The old code was instantiating an `ObjectMapper`, correct?

Reply:
> Seems like the move to instances is not playing nice with SPI, native, and these processors.

That is pretty much expected in native mode.

Reply:
> The old code was instantiating an ObjectMapper, correct?

Yes, recently one, in earlier versions 2 (one for YAML and one for JSON)

Reply:
Gotcha thanks. In that case, I am pretty sure that this general approach augmented with build time discovery of `KubernetesResource` will be fine. 
I'll have a look at the latter tomorrow

Reply:
I don't understand this, actually. If we make it possible for users to inject a mapper, shouldn't that mapper be used here instead of a static version that won't be configured the same way?

Reply:
This code is in the runtime module, therefore accessible to users and I do know users that use this class to create instances of the client so I don't think that this is strictly build time code, even if we'd like to think it is…

Reply:
At the current state, there are no notable differences with what was done before bumping. The logic that was run upstream is now located here. In fact, just running the `scanKubernetesResources()` method from the static block should suffice to fix the native issue.